File: Backend/connecthub/src/main/java/com/example/connecthub/service/CommunityService.java
package com.example.connecthub.service;

import com.example.connecthub.dto.request.CreateCommunityRequest;
import com.example.connecthub.dto.request.UpdateCommunityRequest;
import com.example.connecthub.dto.response.CommunityResponse;

import java.util.List;

public interface CommunityService {

    CommunityResponse createCommunity(
            String email,
            CreateCommunityRequest request);

    List<CommunityResponse> getAllCommunities();

    CommunityResponse getCommunityById(Long id);

    CommunityResponse updateCommunity(
            Long id,
            String email,
            UpdateCommunityRequest request);

    void deleteCommunity(
            Long id,
            String email);

    void joinCommunity(
            Long id,
            String email);

    void leaveCommunity(
            Long id,
            String email);

    com.example.connecthub.dto.response.CommunityJoinRequestResponse createJoinRequest(Long communityId, String email);

    List<com.example.connecthub.dto.response.CommunityJoinRequestResponse> getJoinRequestsForCommunity(Long communityId, String email);

    com.example.connecthub.dto.response.CommunityJoinRequestResponse acceptJoinRequest(Long requestId, String email);

    com.example.connecthub.dto.response.CommunityJoinRequestResponse rejectJoinRequest(Long requestId, String email);

    void cancelJoinRequest(Long communityId, String email);
}

File: frontend/src/services/communityService.js
import client from '../api/client';

export const communityService = {
  getAllCommunities: async () => {
    const response = await client.get('/communities');
    return response.data; // List<CommunityResponse>
  },

  getCommunityById: async (id) => {
    const response = await client.get(`/communities/${id}`);
    return response.data; // CommunityResponse
  },

  createCommunity: async (communityData) => {
    // { name, description, visibility, communityImage }
    const response = await client.post('/communities', communityData);
    return response.data; // CommunityResponse
  },

  updateCommunity: async (id, updateData) => {
    const response = await client.put(`/communities/${id}`, updateData);
    return response.data; // CommunityResponse
  },

  deleteCommunity: async (id) => {
    await client.delete(`/communities/${id}`);
  },

  joinCommunity: async (id) => {
    const response = await client.post(`/communities/${id}/join`);
    return response.data; // String message
  },

  leaveCommunity: async (id) => {
    const response = await client.delete(`/communities/${id}/leave`);
    return response.data; // String message
  },
};
